core/test/utilities/helpers/async-helpers.spec.js

```diff
@@ -5,11 +5,64 @@ const assert = chai.assert;
 import { AsyncHelpers, GenericHelpers } from './../../../src/utilities/helpers';
 
 describe('Async-helpers', () => {
+  let clock;
+  let obj;
   beforeEach(() => {
     // sinon.stub(LuigiConfig, 'getConfigValue');
+    obj = {
+      client: 'here'
+    };
+    clock = sinon.useFakeTimers();
   });
   afterEach(() => {
     //   sinon.restore();
+    clock.restore();
+  });
+
+  describe('waitForKeyExistency', () => {
+    it('returns true if key exists immediately', done => {
+      AsyncHelpers.waitForKeyExistency(obj, 'client').then(result => {
+        assert.equal(result, true);
+        done();
+      });
+      clock.tick(1000);
+    });
+    it('returns true if key exists after some time', done => {
+      obj.client = undefined;
+      AsyncHelpers.waitForKeyExistency(obj, 'client').then(result => {
+        assert.equal(result, true);
+        done();
+      });
+      setTimeout(() => {
+        obj.client = 'yes';
+      }, 500);
+      clock.tick(1000);
+    });
+    it('returns message if key exists after 20000ms', done => {
```

Comment:
we can remove that test, the "if key does not exist" is valid for both cases. maybe rename the one below to "if key does not exist after timeout"